pkg/tsdb/cloudwatch/credentials.go

```diff
@@ -149,16 +149,9 @@ func (e *CloudWatchExecutor) getDsInfo(region string) *DatasourceInfo {
 
 	authType := e.DataSource.JsonData.Get("authType").MustString()
 	assumeRoleArn := e.DataSource.JsonData.Get("assumeRoleArn").MustString()
-	accessKey := ""
-	secretKey := ""
-	for key, value := range e.DataSource.SecureJsonData.Decrypt() {
-		if key == "accessKey" {
-			accessKey = value
-		}
-		if key == "secretKey" {
-			secretKey = value
-		}
-	}
+	decrypted := e.DataSource.DecryptedValues()
+	accessKey := decrypted["accessKey"]
+	secretKey := decrypted["secretKey"]
```

Comment:
This should have a fallback in case they keys aren't present shouldn't it? Might make sense for DecryptedValue to just replace decryptedValue and match its signature.

Reply:
@DanCech no they shouldn't have a fallback - didn't have that before. Or what am I missing here?

Reply:
What happens if those keys aren't present?

Reply:
Okay, guess it makes sense. @aknuds1 can you change this to something like this (as before)
```go 
accessKey := ""
secretKey := ""
if value, exists := decrypted["accessKey"]; exists {
  accessKey = value
}
if value, exists := decrypted["secretKey"]; exists {
  secretKey = value
}
```

Reply:
@marefr this is an api we'll use in every datasource, why not just make decryptedValue public and use those semantics instead of this mess?

Reply:
As discussed in Slack, making the following change doesn't have any practical effect, since the default when looking up a string in a map is the empty string:

```
accessKey := ""
secretKey := ""
if value, exists := decrypted["accessKey"]; exists {
  accessKey = value
}
if value, exists := decrypted["secretKey"]; exists {
  secretKey = value
}
```

Reply:
@DanCech have a look at the comment here https://github.com/grafana/grafana/pull/20579#discussion_r349529035 we now uses `DecryptedValues()` to optimize for not have to lock the cache 2 times and instead 1 time.

Reply:
I'd forgotten about the map semantics, haven't written enough go lately :(  Let's get this out the door!